code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/identity/TokenResourceManager.java

```diff
@@ -57,15 +61,21 @@ private Token createToken(ApiRequest request) {
             throw new ClientVisibleException(ResponseCodes.BAD_REQUEST,
                     "codeInvalid", "Code provided is invalid.", null);
         }
+
+        Identity[] identities = token.getIdentities();
```

Comment:
This is to ensure that any identities that are returned in the token are ones that are meant for the client and not the server.